packages/sdk/utils/src/account.test.ts

```diff
@@ -1,284 +1,499 @@
+        const derivation0 = await generateKeys(mnemonics[i])
+        const derivation1 = await generateKeys(mnemonics[i], undefined, 0, 1)
+        const password = await generateKeys(mnemonics[i], 'password')
+        expect({ derivation0, derivation1, password }).toEqual(expectedPrivateKeys[i])
+      }
+    })
+  })
+
+  describe('.normalizeMnemonic()', () => {
+    it('should normalize phrases with and without accents', async () => {
+      const spanishMnemonics = [
+        'yerno obvio niñez pierna bebé pomelo retorno flujo sacar odio oxígeno rabo', // Correctly accented.
+        'yerno obvio niñez pierna bebé pomelo retorno flujo sacar odio oxigeno rabo', // Missing 1 accent.
+        'yerno obvio niñez pierna bebe pomelo retorno flujo sacar odio oxígeno rabo', // Missing 1 accent.
+        'yerno obvio ninez pierna bebe pomelo retorno flujo sacar odio oxígeno rabo', // Missing 2 accents.
+        'yerno obvio ninez pierna bebe pomelo retorno flujo sacar odio oxigeno rabo', // Missing all 3 accents.
+        'yérno obvio ninez pierña bebe pomelo retorno flujo sacar odio oxigéno rabo', // Incorrect accents.
+      ]
+      const expectedMnemonic = spanishMnemonics[0]
+
+      for (const mnemonic of spanishMnemonics) {
+        expect(normalizeMnemonic(mnemonic)).toEqual(expectedMnemonic)
+      }
+    })
+
+    it('should not normalize accents when the word is from a different language', async () => {
+      // Cases include French mnemonic, missing one accent, and a single Spanish word mixed in, without proper accent.
+      const cases = [
+        {
+          mnemonic:
+            'declarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
+          language: undefined,
+          expected:
+            'déclarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
         },
-      },
-      {
-        derivation0: {
-          address: '0x2bA6d43f0bd3364DD9AF65A80A068a999f311Fd8',
-          privateKey: '4ae237a3fcfcbd877f11cdbf25ea3db398e67c261ddf6448427076da87759b99',
-          publicKey: '03a8c7f50e68fc1c304814ad55837802f0b95b29263cda3d19e86bc26f9eaa5bed',
+        {
+          mnemonic:
+            'declarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
+          language: MnemonicLanguages.french,
+          expected:
+            'déclarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
         },
-        derivation1: {
-          address: '0x773d283771e56abcef5925F9F5Aadb49FB91D647',
-          privateKey: '717dd0b70395964bc4175492f61599e7165cfbc6290fa76798fd77f8dfd9bbb9',
-          publicKey: '0361d8adcac067bb2927d625e642af5f1f53914b102d0740ad97d103ea079a6ce4',
+        {
+          mnemonic:
+            'declarer effrayer estime carbone bebe danger déphaser junior buisson ériger morceau cintrer',
```

Comment:
I'm not sure I understand this test (probably because I don't know French or Spanish very well). The mnemonic is in French but it is expected to be in Spanish, so it adds the accents?

Reply:
Added a comment to explain what this is testing